src/app/header/menu/menu.component.ts

```diff
@@ -1,28 +1,51 @@
-import { Component, OnInit, Input } from '@angular/core';
-import { Router, ActivatedRoute } from '@angular/router';
+import { AfterViewChecked, ChangeDetectorRef, Component, Input, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
 
 import { ContentfulService } from '../../core/contentful.service';
+import { AsmEvent } from '../../core/interfaces/event.interface';
 import { Menu, MenuItem } from '../../core/interfaces/menu.interface';
 
 @Component({
   selector: 'asm-menu',
   templateUrl: './menu.component.html',
   styleUrls: ['./menu.component.scss']
 })
-export class MenuComponent implements OnInit {
+export class MenuComponent implements OnInit, AfterViewChecked {
   @Input() menu: [Menu | MenuItem];
-  event: any;
+  event: AsmEvent;
 
-  constructor(private router: Router,
+  constructor(
     private contentful: ContentfulService,
-    private route: ActivatedRoute) { }
+    private route: ActivatedRoute,
+    private changeDetectionRef: ChangeDetectorRef) { }
 
   ngOnInit(): void {
     this.event = this.route.snapshot.data.event;
   }
 
-  getUrl(item: MenuItem): string {
-    return this.contentful.getUrl(item);
+  ngAfterViewChecked(): void {
+    if (this.menu) {
+      this.menu.forEach((item: MenuItem): void => this.itemHasLink(item));
+    }
+    this.changeDetectionRef.detectChanges();
+  }
+
+  itemHasLink(item: Menu | MenuItem): void {
+    if (item.link) return;
+
+    if (item.__typename === 'MenuItem') {
+      item = item as MenuItem;
+      item.link = this.contentful.getUrl(item);
+    }
+
+    if (item.__typename === 'Menu') {
```

Comment:
You could use here `else if` as if item is `MenuItem` then it can't be `Menu`
```suggestion
    else if (item.__typename === 'Menu') {
```